Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/XMLReader.java

```diff
@@ -0,0 +1,573 @@
+            });
+            return documentBuilder.parse(source).getDocumentElement();
+        } finally {
+            input.close();
+        }
+    }
+
+    /**
+     * Parses an XML DOM tree to a linked data tree.
+     *
+     * @param root
+     *            root node of the XML DOM tree
+     * @param documentNS
+     *            the address of the document, as fall-back namespace for local
+     *            XML elements and attributes
+     * @param storage
+     *            storage to read the XML into
+     * @return the root node of a linked data tree
+     */
+    private static Node toNode(Element root, String documentNS, Storage storage) {
+        return parseNodeElement(root, "", Space.DEFAULT, documentNS, true, storage);
+    }
+
+    /**
+     * Creates a Node from a file.
+     *
+     * @param path
+     *            file to read
+     * @param storage
+     *            storage implementation to read the XML file into
+     * @return the linked data node
+     * @throws SAXException
+     *             if the file is not wellformed
+     * @throws IOException
+     *             if the reading fails
+     */
+    public static Node toNode(File path, Storage storage) throws SAXException, IOException {
+        return toNode(parseXML(new FileInputStream(path), Optional.empty()), globallyUniqueIdentifierForFile(path),
+            storage);
+    }
+
+    /**
+     * Creates a Node from an arbitrary input resource.
+     *
+     * @param bytes
+     *            input data
+     * @param encoding
+     *            character encoding to use
+     * @param documentNS
+     *            the address of the document, as fall-back namespace for local
+     *            XML elements and attributes
+     * @param storage
+     *            storage implementation to read the XML file into
+     * @return the linked data node
+     * @throws SAXException
+     *             if the file is not well-formed
+     * @throws IOException
+     *             if the reading fails
+     */
+    public static Node toNode(InputStream bytes, Optional<String> encoding, String documentNS, Storage storage)
+            throws SAXException, IOException {
+        return toNode(parseXML(bytes, encoding), documentNS, storage);
+    }
+
+    /**
+     * Create a Node from an XML String.
+     *
+     * @param data
+     *            String to parse
+     * @param documentNS
+     *            the address of the document, as fall-back namespace for local
+     *            XML elements and attributes
+     * @param storage
+     *            storage implementation to read the XML file into
+     * @return the created Node
+     * @throws SAXException
+     *             if the XML is semantically wrong
+     */
+    public static Node toNode(String data, String documentNS, Storage storage) throws SAXException {
+        try {
+            return toNode(new ByteArrayInputStream(data.getBytes("UTF-16")), Optional.of("UTF-16"), documentNS,
```

Comment:
You are opening a stream but closing it inside a other method. Is this a good way? Would it not be proper way to open and close a stream inside the same method?